### PR TITLE
Reset Account Management

### LINKED ISSUE #3468: Allow merchants to reset the onboarding progress when onboarding is not completed
Unlike with the prior (OAuth) onboarding flow, the current ("Account Links") flow creates an account at the start of onboarding, and **cannot be restarted** by re-entering the flow (following the recommendation in [this doc](url)).

We've seen a number of reports of the merchant needing to return to an earlier point in the KYC flow (to change country or business type, for instance). We've also seem reports of merchants getting emails from Stripe due to not completing the flow, and asking for the account to be deleted.

We could solve both cases simply by allowing the merchant to delete their existing account, perhaps in some post-onboarding cases as well (see https://github.com/Automattic/woocommerce-payments/issues/2459) but particularly when the onboarding isn't complete, which would restore the prior ability to **start the onboarding anew**.

Here is one proposed way of presenting this action, though it's wide open to design review:

reset-account-mockup

Since a) the button should only show up when there is an existing account, b) the server doesn't currently return any indication for accounts that haven't completed onboarding, and c) it's likely not possible to return such a (non-empty) indication from `GET /accounts` in a backwards-compatible way, we'll just have to decide on some way to encode the not-fully-onboarded state in the server response and have it depend on a plugin version check.

An alternative design that doesn't need this information could be simpler to implement – for instance, a less prominent action (in WooCommerce » Status » Tools?) that just resets the account whether it's already reset or not – but would be less discoverable.

h/t @ricardo for raising this in p1638242085417400-slack-C01BZUL57SQ – some discussion ongoing in that thread.

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #3468

#### Changes proposed in this Pull Request

* Adds Account Tools including reset account management available currently only to partially onboarded accounts (`details_submitted = 0`) 
* Some styling updates to the Fees display to make it the same as the account tools.

Screenshots:

Screenshot 2023-12-21 at 10 02 49

Screenshot 2023-12-21 at 10 02 54

### With Dev Mode Enabled

* Spin up a new JN site of this branch using live branches.
* Using WCPay Dev Tools, ensure Dev Mode is enabled.
* Onboard a new account. When you get to the Stripe KYC, get to the first step after TOS acceptance (First name, last name) and then either close the tab or press 'return to site' to go back to your store.
* Go to the Payments > Overview page. You should **not** be able to see the Account Tools section. 
* Complete the onboarding - the Account Tools should still not be present on the page.

### With Live Mode Enabled

* Spin up a new JN site of this branch using live branches.
* Make sure Dev Tools is not installed, and Dev Mode is disabled.
* Onboard a new account. When you get to the Stripe KYC, get to the first step after TOS acceptance (First name, last name) and then either close the tab or press 'return to site' to go back to your store.
* Go to `Payments`. The account should be restricted and you should be able to see the Account Tools in the account status card.
* Click `Finish Setup`, you should be taken to the Stripe KYC and able to pick up where you left off.
* Click `Reset Account`. First, verify the modal closes via the cancel button and `X` icon.
* Click `Reset Account` and then click the confirmation (red button) on the modal. It should delete the account and take you back to the start of the onboarding MOX. 
* Onboard a new account. At the end of the onboarding, verify the onboarding was successful and the Account Tools are no longer present on the Overview page (because `details_submitted = 1`).
-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7914` or branch name `dev/3468-allow-reset-account-management` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/3468-allow-reset-account-management) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 503f7da735d72048aa42582cce5d009a059f45da
- Build time: 2023-12-21 09:57:51 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/account-status/account-tools/index.tsx
Comment (dmallory42) [inline client/components/account-status/account-tools/index.tsx]: TODO: Uncomment this before merge.

Comment (dmallory42) [reply client/components/account-status/account-tools/index.tsx]: Done 🟢

File: client/components/account-status/account-tools/styles.scss
Comment (oaratovskyi) [inline client/components/account-status/account-tools/styles.scss]: maybe we can also use `$gap` for this which is exactly 16px

File: includes/class-wc-payments-account.php
Comment (oaratovskyi) [inline includes/class-wc-payments-account.php:1052]: I wonder what happens if test_mode is enabled? for dev mode we don't allow it, but for live account in test_mode we do allow deletion, right?

File: client/overview/modal/reset-account/test/__snapshots__/index.test.tsx.snap
Comment (oaratovskyi) [inline client/overview/modal/reset-account/test/__snapshots__/index.test.tsx.snap]: I wonder if this test is valid? seems like there is no reset account modal in this html output

File: client/components/account-status/account-tools/index.tsx
Comment (oaratovskyi) [inline client/components/account-status/account-tools/index.tsx]: could you please review this section and move all the external deps to the section a bit above? 
I noticed @wordpress/components and @wordpress/url positioned in Internal dependencies

**Review by oaratovskyi — COMMENTED** [review]
Hi Dan! It works as expected, except I didn't test the last item:
> Onboard a new account. At the end of the onboarding, verify the onboarding was successful and the Account Tools are no longer present on the Overview page (because details_submitted = 1)

That's due to PO not working due to slots overfill. I'll be happy to test it once it's working 🙂

File: client/components/account-status/account-tools/styles.scss
Comment (dmallory42) [reply client/components/account-status/account-tools/styles.scss]: Thanks, I didn't know that! Updated in 681adad07.

File: includes/class-wc-payments-account.php
Comment (dmallory42) [reply includes/class-wc-payments-account.php:1052]: I don't think it matters for a live account (because of the way the server logic is implemented, test accounts cannot act in live mode, but live accounts can act in either test or live mode).

I also think it's not possible for a live account to access this when in test mode, because the Account Tools only display when the account is not fully onboarded, and the user cannot access the settings to modify the Test Mode value if not fully onboarded.

The server also has some security in place against pretending a live account is actually a test one.

File: client/overview/modal/reset-account/test/__snapshots__/index.test.tsx.snap
Comment (dmallory42) [reply client/overview/modal/reset-account/test/__snapshots__/index.test.tsx.snap]: Good catch! Used a better test in 190a7e145, and it's passing.

File: client/components/account-status/account-tools/index.tsx
Comment (dmallory42) [reply client/components/account-status/account-tools/index.tsx]: Thanks for catching this! Updated in 3d4d66f97.

**Comment by dmallory42** [comment]
Thanks for the review, @oaratovskyi! 

> Hi Dan! It works as expected, except I didn't test the last item:
> 
> >    Onboard a new account. At the end of the onboarding, verify the onboarding was successful and the Account Tools are no longer present on the Overview page (because details_submitted = 1)
> 
> That's due to PO not working due to slots overfill. I'll be happy to test it once it's working 🙂

I tested this scenario and it is working as expected 🙂 :

Screenshot 2023-12-20 at 15 46 17

**Review by oaratovskyi — APPROVED** [review]
Thanks for addressing my feedback! 
Approving this one and have one last question( it's more about the design side of things, so I leave it to @elizaan36, it's more a curiousity for me):

The underline in Reset Account button - should it be there? The buttons look slightly different now, not sure if that's an issue though 
image

**Comment by dmallory42** [comment]
> The underline in Reset Account button - should it be there? The buttons look slightly different now, not sure if that's an issue though

Thanks for spotting it! Addressed by be1ef8f9c8088a062f52456a427cd312ef1b476d.

**Comment by dmallory42** [comment]
@rogermattic because this is my last day before AFK, and you have some AFK, I'm going to go ahead and merge this one without the design review, since I checked it against your Figma designs. The only difference I can see is that the modal styling is slightly different, but I believe this is because the WP component design for a modal changed fairly recently to a more simplified style. Hope that's okay! 

If there are any design quirks in what I implemented then I'm happy to make a follow up PR to address them!

**Comment by oaratovskyi** [comment]
Great work, @dmallory42!

**Comment by rogermattic** [comment]
Great thanks @dmallory42 ! It looks 👌

**Pull request merged**
